### ISSUE TITLE
Pylint considers `__getattr__` when warning on no-member, but not `__getattribute__`

### ISSUE THREAD

**Issue by vmalloc** [PyCQA/pylint#2008]
### Steps to reproduce

The following code issues no "no-member" warning:

```python
class Dynamic(object):

    def __getattr__(self, attr):
        return 'blap'


if __name__ == "__main__":
    obj = Dynamic()
    print(obj.some_attr)

```

While the following does:

```python
class Dynamic(object):

    def __getattribute__(self, attr):
        return 'blap'


if __name__ == "__main__":
    obj = Dynamic()
    print(obj.some_attr)

```

### Current behavior

A "no-member" error is emitted for `some_attr` above

### Expected behavior

No warning should be emitted

### pylint --version output

```
pylint 1.8.3, 
astroid 1.6.2
Python 3.6.5 (default, Mar 30 2018, 06:41:53) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

**Comment by PCManticore** [comment]
Thanks for the issue! This is weird as I cannot reproduce this issue and we already consider `__getattribute__` when we look for `no-member`, as you can see from the method that we use to verify that: https://github.com/PyCQA/astroid/blob/master/astroid/scoped_nodes.py#L2350. Does your initial example look different than the one you posted here?

**Comment by vmalloc** [comment]
Nope, that was the example I used... It's weird since I can't reproduce it either now.. Perhaps was running an older version by mistake. 

Thanks!
